server.py

```diff
@@ -32,7 +33,30 @@ def get_shopcarts(id):
 
     return jsonify(message), rc
 
-
+######################################################################
+# Create a Shopcart
+######################################################################
+@app.route('/shopcarts/<int:id>', methods=['PUT'])
+def create_shopcart(id):
+    """Creates a shopcart and saves it to database
+        POST Request accepts data as: 
+        - id + prod id (sets default to 1)
+        - id + {prod:quant}
+    """
+    # if cart = Shopcart.find(id):
+        # Not clear on what to do here
+    #     return 
```

Comment:
looked up this scenario online, and there are a lot of different opinions, but the most common I saw was to use a `409 Conflict` HTTP response.  If the `uid` already exists, then I believe this should return a `409` with a message detailing that the shopcart already exists. 

https://tools.ietf.org/html/draft-ietf-httpbis-p2-semantics-18#section-7.4.10

But since there are a lot of conflicting opinions, I'm not 100% on this - @thoughts1053 @RegulusXiang what do ya'll think?

Reply:
I'm cool with `409`. There could be multiple ways to do it, but since we're allowing products to come along with the body, this will be the easiest thing.

Page I like to quickly look up status codes: http://www.restapitutorial.com/httpstatuscodes.html

Reply:
So still a last question. Im creating a new shopcart for a given uid so shouldn't the request be 
POST to /shopcarts  instead of  POST to /shopcarts/id  ?? 
OR it does not matter? Because rofrano has a POST to /pets but our issue says /shopcarts/id
@thoughts1053  @radhikamattoo